### DISCUSSION TITLE
generate car on difference start

### DISCUSSION THREAD

**Discussion by LWY2012** [trackmania-rl/tmrl#113] [Ideas]
I think it would be possible generate car on difference start.For example, the first round at the start and the next round to be born at different track locations in each round.I think this would help the AI learn the near end of the track better because this would allow the AI to more fully understand the entire track, rather than just being familiar with the front of its track and not knowing anything about the back of the track.(I always find this happens in my training.)Also I think this may help delay overfitting from occurring because by doing so the AI will be able to collect more data.

**Comment by yannbouteiller** [comment]
Hi, this is what Linesight and Yosh do in TrackMania Nations Forever and what the people at Sony did in Sophy, but I don't think this is possible in TrackMania 2020.

One of the things I like about TrackMania 2020 is that, as far as I know, it is not usable as a simulator with the kind of low-level access that allows you to accelerate or pause the simulation like in Nations Forever. This makes the example TMRL pipeline very close to training on a real robot.

**Reply by LWY2012** [reply]
I think it's also possible to generate cars at different locations in TrackMania 2020, see the "Pipe Maze" section at **https://www.youtube.com/watch?v=kojH8a7BW04**
